### ISSUE TITLE
Cytoscape v4 changes

### ISSUE THREAD

**Issue by maxkfranz** [cytoscape/cytoscape.js#2019]
Brainstorming of improvements that would cause breaking changes for v4

- [ ] `ele.json()` returns `classes` as an array

**Comment by pmackay** [comment]
>ele.json() returns classes as an array

Does this mean that it would _include_ classes as well as other data? 

Is there any way currently to get the list of classes an element has?

**Comment by maxkfranz** [comment]
> Does this mean that it would include classes as well as other data?

`classes: ['foo', 'bar']` versus `classes: 'foo bar'`

> Is there any way currently to get the list of classes an element has?

`ele.json().classes`

**Comment by StevenLangbroek** [comment]
Since a `class` is a concept in Javascript now, maybe it's better to use `classNames`? Ecosystem seems to be standardising on this.

**Comment by maxkfranz** [comment]
Yes, but that wouldn't have to be a breaking (v4) change.  The old key could still be supported.

**Comment by theFrawley** [comment]
Our app would really benefit from Pointer Events.
I believe that web is moving that way - with pointer events now supported in all the main browsers.

Please please please could you add them to Cytoscape.js? :)

**Comment by maxkfranz** [comment]
This issue is for discussing v4 API -- i.e. breaking API changes.  If you have a general feature in mind that is additive w.r.t. API, then feel free to make an ordinary feature request and the corresponding pull request.
